R/two-column-layouts.R

```diff
@@ -214,3 +241,41 @@ gt_two_column_layout <- function(tables = NULL, output = "viewer",
     double_tables
   }
 }
+
+extract_tab_header_and_style <- function(table) {
+  raw_html <- gt::as_raw_html(table, inline_css = FALSE) %>%
+    xml2::read_html()
+  
+  gt_title <- raw_html %>%
+    xml2::xml_find_first("//*[contains(concat(' ',normalize-space(@class),' '),' gt_title ')]")
+  
+  gt_subtitle <- raw_html %>%
+    xml2::xml_find_first("//*[contains(concat(' ',normalize-space(@class),' '),' gt_subtitle ')]")
+  
+  gt_table_id <- raw_html %>%
+    xml2::xml_find_all("//body/div") %>%
+    xml2::xml_attr("id")
+  
+  s <- raw_html %>%
+    xml2::xml_find_first("//style") %>%
+    xml2::xml_contents() %>%
+    xml2::xml_text() %>%
+    gsub(gt_table_id, "mycombinedtable", x = .) %>%
+    gsub("mycombinedtable table", "mycombinedtable div", x = .)
+  
+  list(
+    title = xml_missing(gt_title),
+    title_class = paste("gt_table", xml2::xml_attr(gt_title, "class")),
+    title_style = xml2::xml_attr(gt_title, "style"),
+    subtitle = xml_missing(gt_subtitle),
+    subtitle_class = paste("gt_table", xml2::xml_attr(gt_subtitle, "class")),
+    subtitle_style = xml2::xml_attr(gt_subtitle, "style"),
+    style = s
+  )
+}
+
+xml_missing <- function(xml){
+  xml_txt <- xml2::xml_text(xml)
+  if (is.na(xml_txt)) return(NULL)
```

Comment:
xml2 returns `NA` if the text is missing. We return `NULL` to drop it entirely from the html.